apps/modernization-ui/src/apps/search/event/components/InvestigationSearch/InvestigationGeneralFields.tsx

```diff
@@ -42,7 +42,27 @@ export const InvestigationGeneralFields = ({ form }: InvestigationGeneralAccordi
     ): void => {
         // Clear event id field on deselect
         if (!e.target.value) {
-            form.resetField('eventId.id');
+            form.setValue('eventId', { investigationEventType: undefined } as any, {
+                shouldDirty: true,
+                shouldValidate: true
+            });
+            return;
+        }
+        onChange(e);
+    };
+
+    const handleChangeWithUndefinedDefaultValue = (
+        name: String,
+        e: ChangeEvent<HTMLSelectElement>,
+        onChange: (event: ChangeEvent<HTMLSelectElement>) => void
+    ): void => {
+        // Clear event id field on deselect
+        if (!e.target.value) {
+            form.setValue(name as any, undefined as any, {
+                shouldDirty: true,
+                shouldValidate: true
+            });
```

Comment:
Interesting!  I think we can take this further and allow it to handle empty string selection for all Select Inputs